tracing/src/main/java/com/palantir/remoting3/tracing/Tracer.java

```diff
@@ -45,156 +41,74 @@
 
     private Tracer() {}
 
-    // Thread-safe since thread-local
-    private static final ThreadLocal<Trace> currentTrace = ThreadLocal.withInitial(() -> {
```

Comment:
This is the critical line! remoting3 no longer maintains any state, instead it's all delegated to the shiny new tracing-java.